integration/spark/src/main/scala/org/apache/spark/util/AlterTableUtil.scala

```diff
@@ -442,7 +442,11 @@ object AlterTableUtil {
         // update schema for long string columns
         updateSchemaForLongStringColumns(thriftTable, longStringColumns.get)
       } else if (propKeys.exists(_.equalsIgnoreCase("long_string_columns") && !set)) {
-        updateSchemaForLongStringColumns(thriftTable, "")
+        val varcharColumns = carbonTable.getTableInfo.getFactTable.getListOfColumns.asScala.toList
```

Comment:
here instead of looping all columns, just check whether the long string property present in `tblPropertiesMap`

Reply:
Done